dialogue-core/src/main/java/com/palantir/dialogue/core/CautiousIncreaseAggressiveDecreaseConcurrencyLimiter.java

```diff
@@ -68,18 +72,25 @@
      * {@link Permit#onFailure} which delegate to
      * ignore/dropped/success depending on the success or failure state of the response.
      * */
-    Optional<Permit> acquire() {
-        int currentInFlight = getInflight();
-
-        // We don't want to hand out a permit if there are 4 inflight and a limit of 4.1, as this will immediately send
-        // our inflight number to 5, which is clearly above the limit.  Instead, we wait until there is capacity for
-        // one whole request before handing out a permit. In the worst-case scenario with zero inflight and a limit of
-        // 1, we'll still hand out a permit
-        if (currentInFlight <= getLimit() - 1) {
-            int inFlightSnapshot = inFlight.incrementAndGet();
-            return Optional.of(new Permit(inFlightSnapshot));
+    Optional<Permit> acquire(boolean force) {
+        while (true) {
+            int currentInFlight = getInflight();
+
+            // We don't want to hand out a permit if there are 4 inflight and a limit of 4.1, as this will immediately
+            // send
+            // our inflight number to 5, which is clearly above the limit.  Instead, we wait until there is capacity for
+            // one whole request before handing out a permit. In the worst-case scenario with zero inflight and a limit
+            // of
+            // 1, we'll still hand out a permit
+            if (force || currentInFlight <= getLimit() - 1) {
```

Comment:
This can be simplified
```suggestion
            if (force || currentInFlight < getLimit()) {
```